core/launch_configuration.go

```diff
@@ -0,0 +1,143 @@
+package autospotting
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+type launchConfiguration struct {
+	*autoscaling.LaunchConfiguration
+}
+
+func (lc *launchConfiguration) countLaunchConfigEphemeralVolumes() (int, error) {
+	count := 0
+
+	if lc.BlockDeviceMappings == nil {
+		return 0, fmt.Errorf("Launch configuration has no block device mappings")
+	}
+
+	for _, mapping := range lc.BlockDeviceMappings {
+		if mapping.VirtualName != nil &&
+			strings.Contains(*mapping.VirtualName, "ephemeral") {
+			logger.Println("Found ephemeral device mapping", *mapping.VirtualName)
+			count++
+		}
+	}
+
+	logger.Printf("Launch configuration would attach %d ephemeral volumes",
+		"if available", count)
+
+	return count, nil
+}
+
+func (lc *launchConfiguration) convertLaunchConfigurationToSpotSpecification(
+	baseInstance *instance,
+	instanceType string,
+	az string) *ec2.RequestSpotLaunchSpecification {
```

Comment:
Here we do take the AZ parameter, but as stated earlier we don't check that the compatible instance, is also in the AZ of the ASG.